src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/ConvertRelatedLinksToMultiUrlPicker.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Persistence.Dtos;
+
+namespace Umbraco.Core.Migrations.Upgrade.V_8_0_0
+{
+    public class ConvertRelatedLinksToMultiUrlPicker : MigrationBase
+    {
+        public ConvertRelatedLinksToMultiUrlPicker(IMigrationContext context) : base(context)
+        { }
+
+        public override void Migrate()
+        {
+            var sqlDataTypes = Sql()
+                .Select<DataTypeDto>()
+                .From<DataTypeDto>()
+                .Where<DataTypeDto>(x => x.EditorAlias == "Umbraco.RelatedLinks"
+                                         || x.EditorAlias == "Umbraco.RelatedLinks2");
+
+            var dataTypes = Database.Fetch<DataTypeDto>(sqlDataTypes);
+            var dataTypeIds = dataTypes.Select(x => x.NodeId).ToList();
+
+            var sqlPropertyTpes = Sql()
+                .Select<PropertyTypeDto>()
+                .From<PropertyTypeDto>()
+                .Where<PropertyTypeDto>(x => dataTypeIds.Contains(x.DataTypeId));
+
+            var propertyTypeIds = Database.Fetch<PropertyTypeDto>(sqlPropertyTpes).Select(x => x.Id);
+
+            var sqlPropertyData = Sql()
```

Comment:
This SQL call causes an exception if `propertyTypeIds` is an empty array, a count check needs to be added before this call

Reply:
I'll fix in v8 branch now

Reply:
Duh! Yep, can just exit here if there's no IDs found, thanks!